envoy/README.md

```diff
@@ -0,0 +1,505 @@
+<div align="center">
+<h1>PATH<br/>Authorization & Rate Limiting</h1>
+<img src="https://storage.googleapis.com/grove-brand-assets/Presskit/Logo%20Joined-2.png" alt="Grove logo" width="500"/>
+
+</div>
+<br/>
+
+# Table of Contents <!-- omit in toc -->
+****
+- [1. Overview](#1-overview)
+  - [1.1. Components](#11-components)
+  - [1.2 URL Format](#12-url-format)
+- [2. Quickstart](#2-quickstart)
+- [3. Envoy Proxy](#3-envoy-proxy)
+  - [3.1. Contents](#31-contents)
+  - [3.2. Envoy HTTP Filters](#32-envoy-http-filters)
+  - [3.3. Request Lifecycle](#33-request-lifecycle)
+- [4. Gateway Endpoint Authorization](#4-gateway-endpoint-authorization)
+  - [4.1 JSON Web Token (JWT) Authorization](#41-json-web-token-jwt-authorization)
+  - [4.2 API Key Authorization](#42-api-key-authorization)
+  - [4.3 No Authorization](#43-no-authorization)
+- [5. External Authorization Server](#5-external-authorization-server)
+  - [5.1. Gateway Endpoints gRPC Service](#51-gateway-endpoints-grpc-service)
+  - [5.2. Remote gRPC Auth Server](#52-remote-grpc-auth-server)
+    - [5.2.1. PATH Auth Data Server](#521-path-auth-data-server)
+    - [5.2.2. Example Gateway Endpoint YAML File](#522-example-gateway-endpoint-yaml-file)
+    - [5.2.3. Custom Remote gRPC Server Implementation](#523-custom-remote-grpc-server-implementation)
+- [6. Rate Limiter](#6-rate-limiter)
+  - [6.1. Rate Limit Configuration](#61-rate-limit-configuration)
+  - [6.2. Documentation and Examples](#62-documentation-and-examples)
+- [7. Architecture](#7-architecture)
+
+
+## 1. Overview
+
+This folder contains the `Envoy Proxy configuration` and the `Go External Authorization Server` required for managing authorization and rate limiting in the PATH service.
+
+### 1.1. Components
+
+- **Envoy Proxy**: A proxy server that handles incoming requests, performs auth checks, and routes authorized requests to the PATH service.
+- **External Authorization Server**: A Go/gRPC server that evaluates whether incoming requests are authorized to access the PATH service.
+- **Rate Limiter**: A service that coordinates rate limiting among all services.
+- **Redis**: A database used by the rate limiter to coordinate rate limiting among all services.
+- **PATH Service**: The service that handles requests after they have been authorized.
+- **Remote gRPC Server** *(Implemented by Gateway Operator)*: A server that provides the external authorization server with data on which endpoints are authorized to use the PATH service.
+  - _A basic implementation of the remote gRPC server that loads data from a YAML file is available as a Docker image._
+
+A [docker-compose.yaml](./docker-compose.yaml) file is provided to run all of these services locally.
+
+_[See the Architecture Diagram for more information on how these components interact.](#7-architecture)_
+
+### 1.2 URL Format
+
+When auth is enabled, the required URL format for the PATH service is:
+
+```
+https://<SERVICE_NAME>.<PATH_DOMAIN>/v1/<GATEWAY_ENDPOINT_ID>
```

Comment:
@commoddity Bump @adshmh's previous comment.

Reply:
1. Yes this is accurate to the current state of the PR. I'm not sure that I follow the premise that JWT being supported necessitates updating the doc though. JWTs are passed as the `Authorization` header with the `Bearer ` prefix. Can you elaborate more on why this would require changing the URL?

2. I don't believe that will be a problem. This URL structure is the same as that used in Portal Middleware and we are able to support REST chains like POKT there. 

That said, I believe any changes to the URL structure should be addressed in a subsequent PR. See here for further discussion: https://github.com/buildwithgrove/path/pull/52#discussion_r1859632536

CC @adshmh @Olshansk